Library/Services/Implementation/NfieldQuotaService.cs

```diff
@@ -0,0 +1,58 @@
+﻿using Newtonsoft.Json;
+using Nfield.Infrastructure;
+using Nfield.Models;
+using Nfield.Services;
+using Nfield.Extensions;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Nfield.SDK.Services.Implementation
+{
+    internal class NfieldQuotaService : INfieldQuotaService, INfieldConnectionClientObject
+    {
+        #region Implementation of INfieldConnectionClientObject
+
+        public INfieldConnectionClient ConnectionClient { get; internal set; }
+
+        public void InitializeNfieldConnection(INfieldConnectionClient connection)
+        {
+            ConnectionClient = connection;
+        }
+
+        #endregion
+
+        #region Implementation of INfieldQuotaService
+
+        public Task<IEnumerable<QuotaFrameVersion>> GetQuotaFrameVersionsAsync(string surveyId)
+        {
+            ValidateParams(surveyId);
+
+            return Client.GetAsync(QuotaFrameVersionsApi(surveyId))
+                         .ContinueWith(task => task.Result.Content.ReadAsStringAsync().Result)
+                         .ContinueWith(task => JsonConvert.DeserializeObject<IEnumerable<QuotaFrameVersion>>(task.Result))
+                         .FlattenExceptions();
+        }
+
+        #endregion
+        private INfieldHttpClient Client
+        {
+            get { return ConnectionClient.Client; }
+        }
+
+        private Uri QuotaFrameVersionsApi(string surveyId)
+        {
+            return new Uri(ConnectionClient.NfieldServerUri, $"Surveys/{surveyId}/QuotaVersions");
+        }
+
+        private static void ValidateParams(string surveyId)
+        {
+            if (surveyId == null)
```

Comment:
`string.IsNullOrEmpty`? I guess you can't use that because you want to throw different exceptions?

Reply:
it was done this way in the survey version service so for consistency I did it the same way, can change it if you like

Reply:
I think it's fine like this.